### PR TITLE
Fix handling of errors in mutations

### PR DESCRIPTION
**Pull request by crisu83**

### CODE REVIEW

File: src/Execution/Executor.php [hunk 1]
```diff
@@ -114,8 +114,8 @@ public function execute(): ?array
             $result = $operation->getOperation() === 'mutation'
                 ? $this->executeFieldsSerially($objectType, $rootValue, $path, $fields)
                 : $this->executeFields($objectType, $rootValue, $path, $fields);
-        } catch (\Throwable $ex) {
-            $this->handleError(new ExecutionException($ex->getMessage(), $fields, null, null, null, null, $ex));
+        } catch (ExecutionException $ex) {
```
Comment (Jalle19) [inline src/Execution/Executor.php]: Now \Throwable and other exceptions won't ever be caught?

Comment (crisu83) [reply src/Execution/Executor.php]: This should be correct, `\Throwable`s are already elsewhere wrapped after `ExecutionException`s.

Comment (crisu83) [reply src/Execution/Executor.php]: What we can do is add another `catch` clause, just to be sure though. Good point.

File: src/Execution/Executor.php [hunk 2]
```diff
@@ -217,8 +218,8 @@ public function executeFieldsSerially(
 
         $promise->then(function ($resolvedResults) use (&$finalResults) {
             $finalResults = $resolvedResults ?? [];
-        })->otherwise(function ($ex) {
-            $this->context->addError($ex);
+        })->otherwise(function (ExecutionException $ex) {
```
Comment (hungneox) [inline src/Execution/Executor.php]: What if `$ex` is not `ExecutionException`? Do we lose that exception?

File: src/Execution/Executor.php [hunk 3]
```diff
@@ -379,7 +385,7 @@ protected function executeFields(
                 continue;
             }
 
-            $doesContainPromise = $doesContainPromise || $this->isPromise($result);
+            $doesContainPromise = $doesContainPromise || $result instanceof ExtendedPromiseInterface;
```
Comment (hungneox) [inline src/Execution/Executor.php]: @crisu83 Not sure why you don't like the `isPromise` method, but we use it in ~3 places at least.

File: src/Execution/Executor.php (hunk 2 above)
Comment (Jalle19) [reply src/Execution/Executor.php]: Sorry, clicked the wrong button. I'm also wondering if this is correct?

File: src/Execution/Executor.php (hunk 3 above)
Comment (crisu83) [reply src/Execution/Executor.php]: Because it’s better to use `instanceof` instead of the `@var` comments.

Comment (hungneox) [reply src/Execution/Executor.php]: @crisu83 If you insist, feel free to change, but we actually don't really use it in the code.

File: src/Execution/Executor.php (hunk 2 above)
Comment (crisu83) [reply src/Execution/Executor.php]: You're right, I've fixed this now.

File: src/Execution/Executor.php [hunk 4]
```diff
@@ -115,7 +114,7 @@ public function execute(): ?array
                 ? $this->executeFieldsSerially($objectType, $rootValue, $path, $fields)
                 : $this->executeFields($objectType, $rootValue, $path, $fields);
         } catch (\Throwable $ex) {
-            $this->handleError(new ExecutionException($ex->getMessage(), $fields, null, null, null, null, $ex));
+            $this->handleError(new ExecutionException($ex->getMessage(), null, null, null, null, null, $ex));
```
Comment (hungneox) [inline src/Execution/Executor.php]: @crisu83 not sure why you removed the `$fields`?

**Pull request merged**